### PR TITLE
Update test_period_range.py

### LINKED ISSUE #21793: BUG: PeriodIndex comparisons break on listlike
```
idx = pd.period_range('2016', periods=3, freq='M')

>>> idx == idx.values
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/pandas/core/indexes/period.py", line 107, in wrapper
    other = Period(other, freq=self.freq)
  File "pandas/_libs/tslibs/period.pyx", line 1777, in pandas._libs.tslibs.period.Period.__new__
TypeError: unhashable type: 'numpy.ndarray'
```

### PR DESCRIPTION
**Pull request by jnecus**
Attempting to implement test for resolved issue #21793

- [x] closes #21793

### CODE REVIEW

File: pandas/tests/indexes/period/test_period_range.py [hunk 1]
```diff
@@ -51,6 +51,9 @@ def test_construction_from_period(self):
         result = period_range(start=start, end=end, freq="Q", name="foo")
         tm.assert_index_equal(result, expected)
 
+        result = period_range(start=start, end=end, freq="Q", name="foo")
+        tm.assert_equal(result, result.values)
```
Comment (jreback) [inline pandas/tests/indexes/period/test_period_range.py:55]: can you follow the OP, e.g. 
```
idx = period_range(...)
result = idx == idx.values
assert result.all()
```

don't object to the above check as well

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexes/period/test_period_range.py (hunk 1 above)
Comment (jnecus) [reply pandas/tests/indexes/period/test_period_range.py:55]: Thanks , hope this looks ok?

File: pandas/tests/indexes/period/test_period_range.py [hunk 2]
```diff
@@ -51,6 +51,10 @@ def test_construction_from_period(self):
         result = period_range(start=start, end=end, freq="Q", name="foo")
         tm.assert_index_equal(result, expected)
 
+        idx = period_range(start=start, end=end, freq="Q", name="foo")
```
Comment (mroeschke) [inline pandas/tests/indexes/period/test_period_range.py:54]: Could you add a comment with the github issue number above this?

```
# GH ...
```

File: pandas/tests/indexes/period/test_period_range.py [hunk 3]
```diff
@@ -51,6 +51,11 @@ def test_construction_from_period(self):
         result = period_range(start=start, end=end, freq="Q", name="foo")
         tm.assert_index_equal(result, expected)
 
+        #test for issue #21793
+        idx = period_range(start=start, end=end, freq="Q", name="foo")
+        result = idx == idx.values
+        assert result.all()
```
Comment (mroeschke) [inline pandas/tests/indexes/period/test_period_range.py:57]: Could you construct the expected result and use `tm.assert_numpy_array_equal` instead?

```
result = idx == idx.values
expected = np.array(...)
tm.assert_numpy_array_equal(result, expected)
```

File: pandas/tests/indexes/period/test_period_range.py [hunk 4]
```diff
@@ -51,6 +51,16 @@ def test_construction_from_period(self):
         result = period_range(start=start, end=end, freq="Q", name="foo")
         tm.assert_index_equal(result, expected)
 
+        # test for issue #21793
+        idx = period_range(start=start, end=end, freq="Q", name="foo")
+        result = idx.values
+        expected = np.array([Period('2017Q1', 'Q-DEC'),
```
Comment (mroeschke) [inline pandas/tests/indexes/period/test_period_range.py:57]: We want to be testing the result of the `idx == idx.values` comparison specifically.

Comment (jnecus) [reply pandas/tests/indexes/period/test_period_range.py:57]: Should be changed now, I appreciate your patience :)

File: pandas/tests/indexes/period/test_period_range.py [hunk 5]
```diff
@@ -51,6 +52,16 @@ def test_construction_from_period(self):
         result = period_range(start=start, end=end, freq="Q", name="foo")
         tm.assert_index_equal(result, expected)
 
+        # test for issue # 21793
+        start, end = Period("2017Q1", freq="Q"), Period("2018Q1", freq="Q")
+        expected = date_range(
```
Comment (mroeschke) [inline pandas/tests/indexes/period/test_period_range.py:57]: why is this `expected` variable here?

Comment (jnecus) [reply pandas/tests/indexes/period/test_period_range.py]: expected variable removed (unnecessary)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jnecus

**Pull request merged**